src/mscorlib/src/System/Array.cs

```diff
@@ -976,7 +976,7 @@ public Object Clone()
             if (array==null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
             Contract.EndContractBlock();
-            return BinarySearch<T>(array, 0, array.Length, value, null);
+            return BinarySearch<T>(array, 0, array.Length, value, (IComparer<T>)null);
```

Comment:
That would work for `List<T>` but it is problematic for arrays. You can add it as an extension method for arrays as well but the other overloads are static so it would be rather inconsistent.